spec/support/shared_contexts/cli/when_main_cli_handler_setup_context.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+shared_context 'when main cli handler setup' do
+  let(:main_handler) do
+    inst = Codeowners::Cli::Main.new
+
+    inst.options = options
+    inst.instance_variable_set(:@config, config)
+    inst.instance_variable_set(:@checker, checker)
+    inst.instance_variable_set(:@owners_list_handler, owners_list_handler)
+
```

Comment:
This is not a very common approach probably we can use `expect(cli).to receive_method_chain('config.owner').and_return(default_owner)` on demand instead of assign private variables reinstantiating the classes and so on...